internals/src/num/mod.rs

```diff
@@ -0,0 +1,23 @@
+//! Helpers for integers
+
+/// Extensions for signed integers
+pub trait IntExt {
+    /// the unsigned integer type
+    type Unsigned;
+
+    ///use this to find unsigned absolute if rust version less that 1.51
+    #[cfg(not(rust_v_1_51))]
+    fn unsigned_abs(self) -> Self::Unsigned;
+}
+
+impl IntExt for i64 {
+    type Unsigned = i64;
```

Comment:
This is signed, should be `u64`, right?

Reply:
Heh, derp, good catch.

Kinda interesting that the code works either way.

Reply:
thanks. Wasn't sure if I should have used casting.